documentation for end-user authencation on ingress-gateway (#2243) (#2904)
istio/istio.io: content/docs/tasks/traffic-management/secure-ingress/index.md

        `"""`
    {{< /text >}}

## Configure end-user authentication on ingress gateway

To support end-user authentication, the Istio ingress gateway sets up a JWT

> authentication policy in the `istio-ingressgateway` file. The ingress gateway

Reviewer: ```suggestion
authentication policy in the `istio-ingressgateway` configuration file. The ingress gateway
```

Author: Done.

> rejects the unauthenticated requests and the request can't access the services

Reviewer: Singular plural congruence: requests-request
```suggestion
rejects the unauthenticated requests and they can't access the services
```

Author: Done.

setup the end-user authentication for the Istio ingress gateway.

1. To verify the setup, run the following curl command and confirm a return
   value of 200:

    {{< text bash >}}
    $ curl -I -HHost:httpbin.example.com http://$INGRESS_HOST:$INGRESS_PORT/status/200
    {{< /text >}}

1. Add the policy requiring the end-user JWT authentication for the
   `istio-ingressgateway` service.

    {{< text bash >}}
    $ cat <<EOF | kubectl apply -f -
    apiVersion: "authentication.istio.io/v1alpha1"
    kind: "Policy"
    metadata:
      name: "ingressgateway"
      namespace: istio-system
    spec:
      targets:
      - name: istio-ingressgateway
      origins:
      - jwt:
          issuer: "testing@secure.istio.io"
          jwksUri: "{{< github_file >}}/security/tools/jwt/samples/jwks.json"
      principalBinding: USE_ORIGIN
    EOF
    {{< /text >}}

1. Run the following curl command:

    {{< text bash >}}
    $ curl -I -HHost:httpbin.example.com http://$INGRESS_HOST:$INGRESS_PORT/status/200
    {{< /text >}}

1. Note the return value of 401. Istio returns this error code value because
   the server expects a JWT but the ingress gateway did not provide one:

> 

Reviewer: There is a missing step here. What does the command do? Does it  provide the needed JWT?

Author: Fixed the steps.

    {{< text bash >}}

>     $ HTTP/1.1 401 Unauthorized

Reviewer: Please split commands from output. I cannot tell when one ends and the other begins.

Author: Done.

    content-length: 29
    content-type: text/plain
    date: Mon, 13 Aug 2018 22:33:32 GMT
    server: envoy
    {{< /text >}}

> 1.  Run the following curl command with the valid JWT:

Reviewer: You have two commands following this step and only one of them is a curl command. Please do not omit information on the step because you assume readers will know what you are doing.

Author: Updated the text after moving it to security.

    {{< text bash >}}
    $ TOKEN=$(curl {{< github_file >}}/security/tools/jwt/samples/demo.jwt -s)
    $ curl --header "Authorization: Bearer $TOKEN" -I -HHost:httpbin.example.com http://$INGRESS_HOST:$INGRESS_PORT/status/200
    {{< /text >}}

> 1.  Note the return value of 200. Istio returns a successful code because the

Reviewer: Was the output included above, I think not but I cannot be sure. Is the output what comes next? It is not clear from the way it's written. If that is the case, I suggest:
```suggestion
1.  If the command was successful, Istio returns a successful code because the request had the valid JWT attached. The output looks similar to:
```

Author: Updated the text after moving it to security.
